scripts/create-status-page

```diff
@@ -0,0 +1,284 @@
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+# ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+# The views and conclusions contained in the software and documentation are those
+# of the authors and should not be interpreted as representing official policies,
+# either expressed or implied, of the FreeBSD Project.
+
+
+# Used to generate the status page:
+# https://github.com/camptocamp/c2cgeoportal/wiki/Status
+
+
+import datetime
+import os
+import os.path
+import re
+import subprocess
+import sys
+import urllib.parse
+from typing import Any, Dict, List, Optional
+
+import dateutil.parser
+import markdown
+import requests
+import yaml
+from markdown.extensions.tables import TableExtension
+
+from markdown_table import render
+
+
+class Status:
+    headers: List[str]
+    data: List[List[str]]
+    _row: Optional[List[str]] = None
+
+    def __init__(self, status=None):
+        self.headers = []
+        self.data = []
+
+        if status is None:
+            self.headers = ["Version", "CI status"]
+            return
+
+        md = markdown.Markdown(extensions=[TableExtension()])
+
+        elem = md.parser.parseDocument([s for s in status.split("\n") if s != "" and s[0] != "#" and s[0] != "["])  # type: ignore
+        self._pe(elem.getroot())
+
+        self.data = [r for r in self.data if len([c for c in r if c != None]) > 0]
+        self.headers.append("CI status")
+        for row in self.data:
+            row.append("")
+
+    def _pe(self, elem):
+        if elem.tag == "th":
+            self.headers.append(elem.text)
+        if elem.tag == "tr":
+            self._row = []
+            self.data.append(self._row)
+        if elem.tag == "td":
+            self._row.append(elem.text)  # type: ignore
+        for e in elem.getchildren():
+            self._pe(e)
+
+    def index(self, ver, link):
+        for ind, row in enumerate(self.data):
+            if row[0] == ver:
+                row[0] = link
+                return ind
+        row = ["" for _ in self.headers]
+        row[0] = link
+        self.data.append(row)
+        return len(self.data) - 1
+
+    def render(self):
+        return render(self.headers, self.data)
+
+
+AUTH_HEADER = ""
+
+
+def main():
+    global AUTH_HEADER
+    AUTH_HEADER = "Bearer {}".format(
+        os.environ["GITHUB_TOKEN"]
+        if "GITHUB_TOKEN" in os.environ
+        else subprocess.check_output(["gopass", "gs/ci/github/token/gopass"]).strip().decode()
+    )
+    with open(os.path.join(os.path.dirname(__file__), "status.yaml")) as f:
+        config = yaml.load(f, Loader=yaml.SafeLoader)
+        for elem in config:
+            project(elem)
+
+    print()
+    print("Created with `scripts/create-status-page`")
+
+
+def project(elem):
```

Comment:
This is a huge function, having a docstring could help.
ex:
Example of an element
Example of what is printed out